docs/resource/sequence/high-level-auth-sequences.puml

```diff
@@ -0,0 +1,113 @@
+@startuml
+autonumber
+Title: Alfresco Components / Identity Service High-Level Flows
+
+actor "User's\nBrowser" as UB
+participant "App" as APP
+participant "Identity\nService" as IS
+participant "Alfresco\nComponent\n(ACS, APS)" as ALFC
+database "Alfresco\nComponent\nDatabase" as ALFCDB
+participant "External IdP" as EXT
+participant "LDAP" as LDAP
+
+note right of ALFC
+- X endpoints represent abbreviated flows where either 
+a subsystem is not configured for the scenario
+or an external system does not know about the particular user
+- Note that this is high-level, not all details are shown
+end note
+
+== 1. JWT (Implicit) Authentication Against Identity Service Configured for LDAP ==
+
+ALFC<->LDAP: LDAP User/Group Sync
+ALFC<->ALFCDB
+... LDAP sync completed ...
+UB<->APP: Request page
+UB<->APP: Login initiated
+UB->IS: Request token
+activate IS
+note left of IS
+Identity Service has not
+authenticated this user
+end note
+IS->UB: Return login page
+deactivate IS
+UB->IS: Submit login form
+activate IS
+note left of IS
+username=bsmith
+password=mypass
+end note
+IS<->LDAP: Validate user/pass
+... User is now authenticated with Identity Service ...
+IS->UB: Return token
+note left of IS
+JWT: eyJhbGci...
+end note
+deactivate IS
+... App / Browser has obtained JWT from Identity Service ...
+UB->ALFC: GET /some-api
+note right of UB
+HTTP Headers
+Authorization: Bearer eyJhbGci...
+end note
+activate ALFC
+ALFC<-->IS: Get signing keys
+ALFC->ALFC: Verify JWT
+ALFC<->ALFCDB: User in Context
+ALFC<->ALFC: Authorization/ACLs
+ALFC->UB: Response
+deactivate ALFC
+
+== 2. JWT (Implicit) Authentication Against Identity Service Configured for SAML ==
+
+ALFC<->LDAP: LDAP User/Group Sync
+ALFC<->ALFCDB
+... LDAP sync completed ...
```

Comment:
Is LDAP sync relevant in this context?

Reply:
Yes.  APS 1.x does not support just-in-time user provisioning and I don't believe we're officially supporting that in ACS 6.1 + IS either.

Reply:
That's true but aren't we, by documenting it like this, implying there _must_ be an LDAP provider in a SAML configuration? Or have I misunderstood the message here?

Reply:
I don't think I explained the source of my confusion clearly. So either we consider two flows independently (and in this case we don't need the LDAP user/group sync in a SAML environment) or the 2nd flow happens after the 1st one, in which case the LDAP sync has already happened. Either way, the 3 LDAP-related steps here to me seem redundant.

Reply:
The LDAP sync is between ACS and/or APS so that those components have the users defined locally to instantiate a context for the user represented in the JWT.

Note that in the depicted SAML flow, the Identity Service does not communicate with the LDAP server, so no LDAP User Federation provider has been configured in Keycloak, though admins might wish to set that up in addition to a SAML IdP.

Reply:
> So either we consider two flows independently (and in this case we don't need the LDAP user/group sync in a SAML environment)

Yes, the two distinct flows are separated by the large grey header breaks.

I through them in the same puml and image as I thought it would be more convenient but perhaps it's just more confusing.  Should I separate them into distinct files?

Reply:
Yes, perhaps splitting them into 2 files would make it clearer (plus we could also pull them into the presentation, the SAML workflow in particular is much better than that we currently have)

Reply:
Done